modules/gravatar-hovercards.php

```diff
@@ -149,6 +169,40 @@ function grofiles_get_avatar( $avatar, $author ) {
 				grofiles_gravatars_to_append( $user->ID );
 		}
 	} else if ( isset( $author->comment_type ) ) {
+		if ( $is_amp ) {
+			if ( 1 === preg_match( '/avatar\/([a-zA-Z0-9]+)\?/', $avatar, $email_hash ) ) {
+				$email_hash  = $email_hash[1];
+				$cache_group = 'gravatar_profiles_';
+				$cache_key   = 'gravatar_profile_' . $email_hash;
+
+				$response_body = wp_cache_get( $cache_key, $cache_group );
+				if ( false === $response_body ) {
+					$response = wp_remote_get( 'https://en.gravatar.com/' . $email_hash . '.json' );
+
+					if ( is_array( $response ) && ! is_wp_error( $response ) ) {
+						$response_body = json_decode( $response['body'] );
+						wp_cache_set( $cache_key, $response_body, $cache_group, 60 * MINUTE_IN_SECONDS );
+					}
+				}
+
+				$profile      = $response_body->entry[0];
+				$display_name = $profile->displayName; // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+				$location     = isset( $profile->currentLocation ) ? $profile->currentLocation : ''; // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+				$description  = isset( $profile->aboutMe ) ? $profile->aboutMe : ''; // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+
+				$avatar = '
+					<figure data-amp-lightbox="true">
+						' . $avatar . '
+						<figcaption>
+							' . $display_name . ( ! empty( $location ) ? ' – ' . $location : '' ) . ( ! empty( $description ) ? ' – ' . $description : '' ) . '
```

Comment:
Could we escape those values as we output them, to ensure we're not breaking things with invalid or malicious cached data?